### DISCUSSION TITLE
What is the workflow for adding / updating translations?

### DISCUSSION THREAD

**Discussion by picnixz** [sphinx-doc/sphinx#12249] [General]
note this was origanally an issue to check translations before releasing

see the next comment for the main questions

---

This is an issue to remind us to check the translations before the next release.

Relates: #11626 and #12184.

**Comment by chrisjsewell** [comment]
also linking https://github.com/sphinx-doc/sphinx/issues/11689

yep thanks, as mentioned in https://github.com/sphinx-doc/sphinx/pull/12184#issuecomment-2039556438,
I think there also needs to be greater clarity in the developer docs as to how this works, and what the workflow is.

If a PR adds / modifies a translatable string,
obviously it would not be generally practical to require their authors to provide all the translations; that is delegated to folks on transifex:
see https://github.com/sphinx-doc/sphinx/blob/db4939f00756d16e1ada3af22e183e4492836507/.github/workflows/transifex.yml where the translatable content gets pushed to transifex, thenperiodically the translations get pulled back.
BUT it would be ideal if they are there by the time we make a release.

I don't know currently what the workflow is to check this 🤷

**Comment by chrisjsewell** [comment]
To add some orthogonal thinking, I'm also interested if / how we can make it easier for sphinx extensions to generate translations, particularly HTML themes (to translate button text, etc)

This is tricky; it is highly unlikely that the developer(s) will have access to translators for all these languages, so then what do you do? Is there a "recommended" workflow for dealing with this?
For example, `furo` one of the more widely used themes still has an open issue for this: https://github.com/pradyunsg/furo/issues/434

I actually have some experience with this: https://github.com/executablebooks/sphinx-book-theme/pull/214, see also https://sphinx-book-theme.readthedocs.io/en/stable/contributing/translate.html
Here I used an auto-translation service, to generate the initial translations, then hope if anyone has a problem with them, they can provide PRs to update.

See also https://github.com/pydata/pydata-sphinx-theme/pull/1192, https://pydata-sphinx-theme.readthedocs.io/en/latest/user_guide/i18n.html#internationalization
